src/rust/util/src/cell.rs

```diff
@@ -12,48 +12,44 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Same as `core::sync::SyncUnsafeCell`, which is still in nightly. Can remove once it is stable.
-pub struct SyncUnsafeCell<T: ?Sized> {
+// Needed because otherwise we have undefined references to _critical_section_1_0_acquire/release.
+// This is the official solution:
+// https://github.com/rust-embedded/critical-section/blob/bc0d1736c3b14e340c1b0c29042edb3e6bffe6b8/README.md#undefined-reference-errors
+#[cfg(not(any(feature = "testing", feature = "c-unit-testing")))]
+use cortex_m as _;
```

Comment:
You could just write somethingg like "Link library to provide symbols necessary for critical section even if code isn't called explicitly."

Reply:
Done. Left the link b/c it could be useful.